### PR TITLE
mod:simplify auth

### PR DESCRIPTION
**Pull request by shiyuhang0**
### What problem does this PR solve? 
Can solve the two problems mentioned in https://github.com/pingcap/tispark/issues/2200

### What is changed and how it works?
1.put `e.injectResolutionRule(new TiAuthRuleFactory(getOrCreateTiContext))` in front of `e.injectResolutionRule(new TiDDLRuleFactory(getOrCreateTiContext))`. so that TiAuthorizationRule can work without catalogplugin because logical plan will't be replaced (happens in TiDDLRule)
2.some codes about auth are deleted in `org.apache.spark.sql.execution.command.tables` and `org.apache.spark.sql.execution.command.databases` because auth will be excute in TiAuthorizationRule. Delete the bug    which is described in the issue as well 

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)

Code changes
 - Has interface methods change
 

Related pr
https://github.com/pingcap/tispark/pull/2185

### CODE REVIEW

**Comment by shiyuhang0** [comment]
@sunxiaoguang @xuanyu66 @birdstorm PTAL

**Comment by xuanyu66** [comment]
`simplify` typo

**Comment by xuanyu66** [comment]
LGTM

File: core/src/test/scala/com/pingcap/tispark/auth/TiAuthIntegrationSuite.scala [hunk 1]
```diff
@@ -94,12 +94,7 @@ class TiAuthIntegrationSuite extends SharedSQLContext {
   test("Use database and select without privilege should not be passed") {
     the[SQLException] thrownBy spark.sql(
       f"use $databaseWithPrefix") should have message s"Access denied for user $user@% to database ${databaseWithPrefix}"
-    if (catalogPluginMode) {
-      an[AnalysisException] should be thrownBy spark.sql(s"select * from $table")
-    } else {
-      the[SQLException] thrownBy spark.sql(
-        s"select * from $table") should have message s"SELECT command denied to user $user@% for table $databaseWithPrefix.$table"
-    }
+    an[AnalysisException] should be thrownBy spark.sql(s"select * from $table")
```
Comment (xuanyu66) [inline core/src/test/scala/com/pingcap/tispark/auth/TiAuthIntegrationSuite.scala:97]: Can we check the message of AnalysisException

**Review by birdstorm — APPROVED** [review]
lgtm

File: core/src/test/scala/com/pingcap/tispark/auth/TiAuthIntegrationSuite.scala (hunk 1 above)
Comment (shiyuhang0) [reply core/src/test/scala/com/pingcap/tispark/auth/TiAuthIntegrationSuite.scala:97]: good idea

**Comment by shiyuhang0** [comment]
@sunxiaoguang cc can the pr be merged?

**Comment by shiyuhang0** [comment]
@birdstorm can you help merge this pr ?

**Pull request merged**